ODC-7803: Add custom icons in import flow

logonoff opened the pull request:
addresses https://issues.redhat.com/browse/ODC-7803 and https://issues.redhat.com/browse/RFE-5057

This PR introduces a new annotation for Deployments and DeploymentConfigs: `app.openshift.io/custom-icon`

This annotation comprises a valid URI pointing to an image that the browser can parse. 

When an icon is uploaded, the icon is stored as a base64-encoded data URI. Due to annotation size restrictions in kubernetes, this means uploaded icons have a heavily restricted maximum file size.

Icons should have a transparent background and should be provided in an 1:1 aspect ratio to ensure proper rendering, however this is not validated against.

This annotation takes precedence over the `app.openshift.io/runtime` label when deciding which icon to render.

https://github.com/user-attachments/assets/206636a7-d240-41ca-b575-8754038940f6

logonoff commented:
docs review:
/assign @opayne1

logonoff commented:
/hold 

UI is WIP and pending testing and refinement

opayne1 commented:
/label docs-approved

andrew-ronaldson commented:
@logonoff can the file path field have pull the status variants. It shows the helper text with the error but the field doesn't change 
<img width="785" alt="Screenshot 2025-07-09 at 10 29 23 AM" src="https://github.com/user-attachments/assets/f06363a5-96a9-49bf-9f02-955c4275eb5a" />

logonoff commented:
/unhold

sanketpathak commented:
@logonoff It shows openshift icon(default icon) in the edit form even with the new icon

https://github.com/user-attachments/assets/c0f2dd1c-3873-450e-9082-24d814569a1a

When a user selects an image, clicks cancel, and then opens the upload icon window, the old selection remains until the user removes it with the close icon. Is this an expected behaviour?

https://github.com/user-attachments/assets/f24d905e-e36e-4a3b-9e3f-9d62bbb9acfd

vikram-raj commented:
/retest

sanketpathak commented:
Thanks @logonoff for fixing the issue mentioned above. The serverless is having issues with the custom icon upload process:

- It does not show the custom icon with the creation process after the build passes 
- Editing the serverless deployment(form view), do not add the custom icon

Although adding an annotation of a custom icon seems to work, which adds the custom image to the serverless revision


https://github.com/user-attachments/assets/0fe317e5-7eb3-4d9d-a64f-c2c073481436

logonoff commented:
> Thanks @logonoff for fixing the issue mentioned above. The serverless is having issues with the custom icon upload process:
> 
>     * It does not show the custom icon with the creation process after the build passes
> 
>     * Editing the serverless deployment(form view), do not add the custom icon

I have fixed this: 

https://github.com/user-attachments/assets/fa7ab301-4b16-4418-9146-d4e19f600906

Note that only for this recording I changed the dropdown to also show the true state of `title`--this is a known bug with the dropdown menu which I am working on fixing in #15284

sanketpathak commented:
@logonoff With deployment, we get the option to make it serverless from the action/context menu. The custom icon is not transferrable to the Make Serverless form. We need to reupload it in the form.

sanketpathak commented:
Thanks @logonoff for the changes. Verified it in a cluster, works fine

https://github.com/user-attachments/assets/f88239db-2723-41f8-a470-66ab446a198d

/label qe-approved

jhadvig commented:
@vikram-raj looks like the timestamp tests are failing in the frontend CI job.
